### DISCUSSION TITLE
여러 캐시된 데이터를 무효화 시키기 위해서 promise.all() 사용하는지에 대해서!

### DISCUSSION THREAD

**Discussion by perfumeplaylist** [bucket-back/bucket-back-frontend#147] [Q&A]
promise.all를 사용하는거에 대해서 결정하는 가장 큰 이유는 **병렬로 처리해서 동시에 처리하는 유무**입니다.
promise.all은 **병렬로 처리해서 여러 캐시된 데이터를 무효화를 동시에 처리하고 모든 쿼리가 무효화될 때까지 기다릴 수 있습니다.**

하지만, 여러 캐싱을 무효화하는 것이 동시에 이루어져야 하는 것이 아니라면, 각각의 쿼리를 무효화하는 것도 가능합니다. 이 경우, 각 쿼리는 독립적으로 무효화되며, 이전 쿼리의 무효화가 완료되기를 기다리지 않습니다.
따라서, 여러 캐싱을 동시에 무효화해야 하는지, 아니면 각각 독립적으로 무효화해도 되는지에 따라 Promise.all()을 사용할지, 아니면 각각의 쿼리를 무효화할지를 결정할 수 있습니다.

**Comment by GBAJS754** [comment]
레일 예시 미쳐ㅆ따
